search_service/models/table.py

```diff
@@ -8,38 +8,46 @@
 
 from .base import Base
 from search_service.models.tag import Tag
+import time
 
 
 @attr.s(auto_attribs=True, kw_only=True)
 class Table(Base):
     """
     This represents the part of a table stored in the search proxy
     """
-    database: str
-    cluster: str
-    schema: str
-    name: str
-    key: str
+    id: str
+    database: Optional[str] = None
```

Comment:
why are we changing them to optional str?

Reply:
some fields may be `None` in elasticsearch, for these fields, required str will fail this method: `data = self.schema(many=True, strict=False).loads(args.get('data')).data`. But yeah, in our use case, I think `database, cluster, schema, table` are required